client/app/scripts/charts/nodes-chart.js

```diff
@@ -82,8 +81,7 @@ class NodesChart extends React.Component {
     state.height = nextProps.forceRelayout ? nextProps.height : (state.height || nextProps.height);
     state.width = nextProps.forceRelayout ? nextProps.width : (state.width || nextProps.width);
 
-    // _.assign(state, this.updateGraphState(nextProps, state));
-    if (nextProps.forceRelayout || !isSameTopology(nextProps.nodes, this.props.nodes)) {
+    if (nextProps.forceRelayout || nextProps.nodes !== this.props.nodes) {
```

Comment:
Two things are getting mixed up here. Generally, in this class `props.nodes` have to become `state.nodes`, otherwise the changes wont be rendered.

This section makes sure not to run the layout engine code if the topo has not changed, it does create `state.nodes` along the way. (I guess this is bad design.)

What you need is `props.nodes`' properties being merged into `state.nodes` if `props.nodes` and `nextProps.nodes` are not the same. This should be a separate step.